### PR TITLE
adds `findfs` for ROOT and `while :;` time limit

### PR DESCRIPTION
**Pull request by shadichy**
I used builtin busybox's findfs to parse UUID=... and LABEL=... into real (existing) block devices
Also,  I've modified the ROOT variable to be capable of accepting filesystem's uuid or label as valid block/sysroot device,  and added SYSTEM as an optional variable for specifying system image/block device

### CODE REVIEW

**Comment by hmtheboy154** [comment]
Can you re-do it but only with the main changes, not with the minor spacing and edit
That should be a separate PR

File: initrd/init [hunk 1]
```diff
@@ -337,19 +294,62 @@ for c in `cat /proc/cmdline`; do
...
-		# Also umount /android and /sfs if already mounted
-		mountpoint -q /android && umount /android
-		mountpoint -q /sfs && umount /sfs
-		mountpoint -q /mnt && umount /mnt
-	done
-	sleep 1
-	echo -n .
+# if the android is partition table-based
+IS_ALL_BLOCKDEV=1
+for dev in $SYSTEM $DATA $RECOVERY; do
```
Comment (hmtheboy154) [inline initrd/init:303]: recovery is a gzip cpio image just like initrd.img so there's no `$RECOVERY` option

forgot to tell you that

File: initrd/init [hunk 2]
```diff
@@ -337,19 +294,62 @@ for c in `cat /proc/cmdline`; do
...
+	if [ -z "$ROOT" ]; then
+		source /hooks/0-find-rootdev
+		find_rootdev 10 || debug_shell debug-notfound # maximum 10 iterations
+	else
+		try_mount ro $dev /mnt || debug_shell debug-mountrootfail
+	fi
+
+	if [ -n "$INSTALL" ]; then
+		if [ ! -d "/mnt/isolinux" ] || [ ! -f "/mnt/install.img" ]; then
+			debug_shell debug-missinginstall
```
Comment (hmtheboy154) [inline initrd/init]: What's `debug-missinginstall` ?

File: initrd/init [hunk 3]
```diff
@@ -337,19 +294,62 @@ for c in `cat /proc/cmdline`; do
...
+
+	if [ -n "$iso" -a -e /mnt/$iso ]; then
+		mount --move /mnt /iso
+		mkdir /mnt/iso
+		mount -o loop /iso/$iso /mnt/iso
+	fi
+
+	if [ -z "$SYSTEM" ]; then
+		source /hooks/1-find-system
+		find_system || debug_shell debug-mountsystemfail
```
Comment (hmtheboy154) [inline initrd/init]: Also this

**Comment by hmtheboy154** [comment]
My current recommendation is that :
- Check the 17 branch of newinstaller for Recovery port, so you can know what to do
https://github.com/BlissOS/bootable_newinstaller/commit/bc4dcb69f0f8591b67de5827d389fbacc0eb214e

I will backport this down to older Android soon once I'm done.

- Actually test and boot to see if it's working, you can just unpack and edit initrd.img

**Review by hmtheboy154 — CHANGES_REQUESTED** [review]

File: initrd/init (hunk 1 above)
Comment (shadichy) [reply initrd/init:303]: I already know
This is for cases like partition table-based installation so either use another `init` or create some other variables works

File: initrd/init (hunk 2 above)
Comment (shadichy) [reply initrd/init]: At first these 5 lines was in `check_root` function but as I modified it these got moved out of the function
It checks if INSTALL is specified but whether install image exist or not. If not, it launches the interactive shell for user to manually fix the problem

File: initrd/init (hunk 3 above)
Comment (shadichy) [reply initrd/init]: Same almost. If system.?fs, system_$SLOT.img or system is not found under ($ROOT)/$SRC, then drop a shell for the user to solve (both in most cases wont likely to happen)

File: initrd/init (hunk 2 above)
Comment (hmtheboy154) [reply initrd/init]: but then there should be a `debug-missinginstall`  function somewhere right ? I don't see it anywhere, same with `debug-mountsystemfail`

Comment (shadichy) [reply initrd/init]: Debug message; not implemented

Comment (hmtheboy154) [reply initrd/init]: either implement or remove it

**Comment by hmtheboy154** [comment]
wtf

**Comment by shadichy** [comment]
> wtf

Wrong repo, sorry
Will make another pr

**Pull request closed without merging**